### PR TITLE
fix(hosts): extra-host params should be 'domain:ip'

### PR DESCRIPTION
**Pull request by tosone**

### CODE REVIEW

File: pkg/composer/serviceparser/serviceparser.go [hunk 1]
```diff
@@ -500,8 +500,8 @@ func newContainer(project *types.Project, parsed *Service, i int) (*Container, e
 			c.RunArgs = append(c.RunArgs, fmt.Sprintf("-e=%s=%s", k, *v))
 		}
 	}
-	for _, v := range svc.ExtraHosts {
-		c.RunArgs = append(c.RunArgs, fmt.Sprintf("--add-host=%s", v))
+	for k, v := range svc.ExtraHosts {
```
Comment (manugupt1) [inline pkg/composer/serviceparser/serviceparser.go:503]: Nice catch! thanks!
This is the docs: https://docs.docker.com/compose/compose-file/compose-file-v2/

**Review by manugupt1 — APPROVED** [review]

File: pkg/composer/serviceparser/serviceparser.go [hunk 2]
```diff
@@ -500,8 +500,8 @@ func newContainer(project *types.Project, parsed *Service, i int) (*Container, e
...
+		c.RunArgs = append(c.RunArgs, fmt.Sprintf("--add-host=\"%s:%s\"", k, v))
```
Comment (junnplus) [inline pkg/composer/serviceparser/serviceparser.go:504]: why do we need quotes?

Comment (tosone) [reply pkg/composer/serviceparser/serviceparser.go:504]: Because the colon should be escaped, sometimes it will be treated as an operator.

**Review by junnplus — APPROVED** [review]
LGTM

**Review by Zheaoli — APPROVED** [review]
LGTM

**Comment by tosone** [comment]
@AkihiroSuda Could you please help me merge this pr? ^_^

File: pkg/composer/serviceparser/serviceparser.go (hunk 2 above)
Comment (AkihiroSuda) [reply pkg/composer/serviceparser/serviceparser.go:504]: What is an “operator” here

**Comment by AkihiroSuda** [comment]
@tosone 
ping https://github.com/containerd/nerdctl/pull/1267#discussion_r933314875

File: pkg/composer/serviceparser/serviceparser.go (hunk 2 above)
Comment (tosone) [reply pkg/composer/serviceparser/serviceparser.go:504]: https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion, I have tested it in my dev. I have many apps startup with `nerdctl compose`.

Comment (AkihiroSuda) [reply pkg/composer/serviceparser/serviceparser.go:504]: I don’t think we are using bash here

Comment (tosone) [reply pkg/composer/serviceparser/serviceparser.go]: seems ok

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**